File: frpc/csharp-peer/Models/Echo.cs
using Fomoxa;

namespace Fomoxa.Rpc.Demo.Models
{
    [Network]
    [Codec("rpc")]
    public class EchoRequest
    {
        [Network("string")]
        [Codec("rpc")]
        public string Text { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class EchoResponse
    {
        [Network("string")]
        [Codec("rpc")]
        public string Text { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class CountRequest
    {
        [Network("u32")]
        [Codec("rpc")]
        public uint Count { get; set; }
    }

    [Network]
    [Codec("rpc")]
    public class CountItem
    {
        [Network("u32")]
        [Codec("rpc")]
        public uint Value { get; set; }
    }
}

File: frpc/rust-peer/src/demo/src/models/echo.rs
#[allow(unused_imports)]
use fomoxa_attributes::{codec, network};

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct EchoRequest {
    #[network(string)]
    #[codec(rpc)]
    pub text: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct EchoResponse {
    #[network(string)]
    #[codec(rpc)]
    pub text: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct CountRequest {
    #[network(u32)]
    #[codec(rpc)]
    pub count: u32,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct CountItem {
    #[network(u32)]
    #[codec(rpc)]
    pub value: u32,
}
